src/Symfony/Component/Process/Process.php

```diff
@@ -397,15 +397,16 @@ public function restart(callable $callback = null, array $env = []): static
      * from the output in real-time while writing the standard input to the process.
      * It allows to have feedback from the independent process during execution.
      *
-     * @param callable|null $callback A valid PHP callback
+     * @param callable|null $callback        A valid PHP callback
+     * @param callable|null $waitingCallback A valid PHP callback triggered while waiting
      *
      * @return int The exitcode of the process
      *
      * @throws ProcessTimedOutException When process timed out
      * @throws ProcessSignaledException When process stopped after receiving signal
      * @throws LogicException           When process is not yet started
      */
-    public function wait(callable $callback = null): int
+    public function wait(callable $callback = null, callable $waitingCallback = null): int
```

Comment:
Adding a new parameter (even an optional one) to a non-final method is a breaking change. You need to comment out the parameter and use `func_num_args()`/`func_get_arg()` to retrieve it.